Description: Build supertuxkart against system enet instead of embedded copy
Forwarded: not-yet
Author: Vincent Cheng <vcheng@debian.org>
Last-Update: 2015-05-13

Index: supertuxkart-0.9.1/CMakeLists.txt
===================================================================
--- supertuxkart-0.9.1.orig/CMakeLists.txt	2015-10-19 19:32:25.150024146 +0200
+++ supertuxkart-0.9.1/CMakeLists.txt	2015-10-19 19:32:57.026612596 +0200
@@ -62,9 +62,9 @@
 add_subdirectory("${PROJECT_SOURCE_DIR}/lib/bullet")
 include_directories("${PROJECT_SOURCE_DIR}/lib/bullet/src")
 
-# Build the ENet UDP network library
-add_subdirectory("${PROJECT_SOURCE_DIR}/lib/enet")
-include_directories("${PROJECT_SOURCE_DIR}/lib/enet/include")
+# ENet
+find_package(ENet REQUIRED)
+include_directories(${ENET_INCLUDE_DIR})
 
 # Build glew library
 add_subdirectory("${PROJECT_SOURCE_DIR}/lib/glew")
@@ -331,11 +331,11 @@
     bulletdynamics
     bulletcollision
     bulletmath
-    enet
     glew
     stkirrlicht
     ${Angelscript_LIBRARIES}
     ${CURL_LIBRARIES}
+    ${ENET_LIBRARIES}
     ${OGGVORBIS_LIBRARIES}
     ${OPENAL_LIBRARY}
     ${OPENGL_LIBRARIES}
Index: supertuxkart-0.9.1/cmake/FindENet.cmake
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ supertuxkart-0.9.1/cmake/FindENet.cmake	2015-10-19 19:32:25.146024073 +0200
@@ -0,0 +1,22 @@
+# - Find ENet
+# Find the ENet includes and libraries
+#
+# Following variables are provided:
+# ENET_FOUND
+#     True if ENet has been found
+# ENET_INCLUDE_DIR
+#     The include directories of ENet
+# ENET_LIBRARIES
+#     ENet library list
+
+
+find_path(ENET_INCLUDE_DIR enet/enet.h /usr/include)
+find_library(ENET_LIBRARY NAMES enet PATHS /usr/lib)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(ENet DEFAULT_MSG ENET_INCLUDE_DIR ENET_LIBRARY)
+
+# Publish variables
+set(ENET_INCLUDE_DIRS ${ENET_INCLUDE_DIR})
+set(ENET_LIBRARIES ${ENET_LIBRARY})
+mark_as_advanced(ENET_INCLUDE_DIR ENET_LIBRARY)
